### ISSUE TITLE
RegularGridData needs positive spacing

### ISSUE THREAD

**Issue by evetion** [JuliaEarth/GeoStatsBase.jl#21]
I encountered this in https://github.com/evetion/GeoArrays.jl/blob/master/src/interpolate.jl#L12, where I use the spacing available in the `GeoArray`. In the case of GDAL created geotiffs, these are always negative for the y direction.

Is there a reason why we don't support negative spacing? In the meantime I guess I can force the spacing to be positive (just taking the `abs`) for both the `RegularGridData` and the `RegularGrid`, which should produce the same results.

**Comment by juliohm** [comment]
That is an interesting question. It seems natural at first to allow negative spacing to mirror the grid, but this can have major impacts in the assumptions we've made in the algorithms. For example, we can always compute the volume of a grid cell when the spacing is positive by simply raising the spacing to the dimension s^N of the grid. In 3D, the formula would produce negative volume for the cell if one of the spacing is negative. This is a simple example, but maybe there are other more advanced used cases that would be more complicated to implement if we allow negative spacing. What do you think?

I wonder if the origin of the GDAL-loaded grid could be shifted to the to lower left origin of the grid without penalties? I don't know to what extent it is easy to generalize the origin of the RegularGrid to some other reference corner without breaking algorithms.

**Comment by evetion** [comment]
Shifting is possible, but needs a flip of the array itself, which can be costly. Since I use it purely for interpolation, I've made the spacing always positive in https://github.com/evetion/GeoArrays.jl/commit/9b16e9e7c40a18d755afb10fae5c7645b85ec7b2#diff-fca3de692a5cfd93374010b5667e1829R12. This shouldn't affect the interpolation itself, as the distances, neighbors etc. remain the same.

I agree that negative spacing could and will present problems in algorithms. Especially if some implementations assumes positive spacing but quietly allows for negative ones because we changed it here. So let's keep this as it is now.
